[MSE] add track id test for MSE

jyavenard opened the pull request:
#### 65d8c44cce8612965ea93adb40dc64bf1c965cf4
<pre>
[MSE] add track id test for MSE
<a href="https://bugs.webkit.org/show_bug.cgi?id=265935">https://bugs.webkit.org/show_bug.cgi?id=265935</a>
<a href="https://rdar.apple.com/119244009">rdar://119244009</a>

Reviewed by Youenn Fablet.

Test exposed bug missed by bug 265718, where HashMap&lt;uint64_t&gt; where left in the code,
which caused an assertion due to the use of forbidden value. Replace them with StdUnorderedMap.

Fly-by: move track tests from media to media/track

* LayoutTests/media/track/media-audio-track-expected.txt: Copied from LayoutTests/media/media-audio-track-expected.txt.
* LayoutTests/media/track/media-audio-track.html: Renamed from LayoutTests/media/media-audio-track.html.
* LayoutTests/media/track/media-source-audio-track-expected.txt: Renamed from LayoutTests/media/media-audio-track-expected.txt.
* LayoutTests/media/track/media-source-audio-track.html: Added.
* LayoutTests/media/track/media-video-audio-track-expected.txt: Renamed from LayoutTests/media/media-video-audio-track-expected.txt.
* LayoutTests/media/track/media-video-audio-track.html: Renamed from LayoutTests/media/media-video-audio-track.html.
* LayoutTests/platform/glib/TestExpectations:
* LayoutTests/platform/mac-wk1/TestExpectations:
* Source/WebCore/platform/graphics/avfoundation/objc/SourceBufferPrivateAVFObjC.h:
* Source/WebCore/platform/graphics/avfoundation/objc/SourceBufferPrivateAVFObjC.mm:
(WebCore::SourceBufferPrivateAVFObjC::precheckInitialisationSegment):
(WebCore::SourceBufferPrivateAVFObjC::didProvideMediaDataForTrackId):
(WebCore::SourceBufferPrivateAVFObjC::processFormatDescriptionForTrackId):
(WebCore::SourceBufferPrivateAVFObjC::didProvideContentKeyRequestInitializationDataForTrackID):
(WebCore::SourceBufferPrivateAVFObjC::appendInternal):
(WebCore::SourceBufferPrivateAVFObjC::destroyRenderers):
(WebCore::SourceBufferPrivateAVFObjC::clearTracks):
(WebCore::SourceBufferPrivateAVFObjC::trackDidChangeEnabled):
(WebCore::SourceBufferPrivateAVFObjC::setCDMInstance):
(WebCore::SourceBufferPrivateAVFObjC::flush):
(WebCore::SourceBufferPrivateAVFObjC::rendererWasAutomaticallyFlushed):
(WebCore::SourceBufferPrivateAVFObjC::trackIsBlocked const):
(WebCore::SourceBufferPrivateAVFObjC::canEnqueueSample):
(WebCore::SourceBufferPrivateAVFObjC::enqueueSample):
(WebCore::SourceBufferPrivateAVFObjC::isReadyForMoreSamples):
(WebCore::SourceBufferPrivateAVFObjC::didBecomeReadyForMoreSamples):
(WebCore::SourceBufferPrivateAVFObjC::notifyClientWhenReadyForMoreSamples):

Canonical link: <a href="https://commits.webkit.org/271655@main">https://commits.webkit.org/271655@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/06838efc72565ac3ce6ec284cad51f99bc14fee2

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/29104 "3 style errors") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/7772 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/30441 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/31703 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/26491 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/29700 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/9953 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/5078 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/26501 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/29376 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/47/builds/6462 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/24942 "Passed tests") | [❌ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/5568 "Found 1 new API test failure: /WPE/TestWebsiteData:/webkit/WebKitWebsiteData/service-worker-registrations (failure)") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/42/builds/5723 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/25971 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/33042 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/26616 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/26408 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/31938 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/5677 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/3845 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/29722 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/7342 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/25767 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/6955 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/6181 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/6198 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

youennf commented on Source/WebCore/platform/graphics/avfoundation/objc/SourceBufferPrivateAVFObjC.mm:
```diff
@@ -1283,10 +1289,12 @@ static bool sampleBufferRenderersSupportKeySession()
-        if (RefPtr player = this->player(); player && !sample->isNonDisplaying())
-            player->setHasAvailableAudioSample(renderer.get(), true);
+        if (auto it = m_audioRenderers.find(trackID); it != m_audioRenderers.end()) {
+            auto& renderer = it->second;
```
s/auto&/RetainPtr/

youennf commented on Source/WebCore/platform/graphics/avfoundation/objc/SourceBufferPrivateAVFObjC.mm:
```diff
@@ -816,16 +819,19 @@ static bool sampleBufferRenderersSupportKeySession()
+        if (auto it = m_audioRenderers.find(trackID); it != m_audioRenderers.end()) {
 ALLOW_NEW_API_WITHOUT_GUARDS_BEGIN
-        RetainPtr<AVSampleBufferAudioRenderer> renderer = m_audioRenderers.get(trackID);
+            RetainPtr<AVSampleBufferAudioRenderer> renderer = it->second;
```
Can we just use `RetainPtr renderer`?

youennf commented on Source/WebCore/platform/graphics/avfoundation/objc/SourceBufferPrivateAVFObjC.mm:
```diff
@@ -716,7 +716,8 @@ static bool sampleBufferRenderersSupportKeySession()
 
-    for (auto& renderer : m_audioRenderers.values()) {
+    for (auto& pair : m_audioRenderers) {
+        auto& renderer = pair.second;
```
RetainPtr?

jyavenard replied:
I can understand replacing auto for Ref/RefPtr/RetainPtr for clarity, but here the data is owned by the map, we don't need refcount churn.

youennf commented on Source/WebCore/platform/graphics/avfoundation/objc/SourceBufferPrivateAVFObjC.mm:
```diff
@@ -737,14 +738,16 @@ static bool sampleBufferRenderersSupportKeySession()
 {
-    for (auto& track : m_videoTracks.values()) {
+    for (auto& pair : m_videoTracks) {
+        auto& track = pair.second;
```
Ref here and below?

jyavenard committed:
[MSE] add track id test for MSE
https://bugs.webkit.org/show_bug.cgi?id=265935
rdar://119244009

Reviewed by Youenn Fablet.

Test exposed bug missed by bug 265718, where HashMap<uint64_t> where left in the code,
which caused an assertion due to the use of forbidden value. Replace them with StdUnorderedMap.

Fly-by: move track tests from media to media/track

* LayoutTests/media/track/media-audio-track-expected.txt: Copied from LayoutTests/media/media-audio-track-expected.txt.
* LayoutTests/media/track/media-audio-track.html: Renamed from LayoutTests/media/media-audio-track.html.
* LayoutTests/media/track/media-source-audio-track-expected.txt: Renamed from LayoutTests/media/media-audio-track-expected.txt.
* LayoutTests/media/track/media-source-audio-track.html: Added.
* LayoutTests/media/track/media-video-audio-track-expected.txt: Renamed from LayoutTests/media/media-video-audio-track-expected.txt.
* LayoutTests/media/track/media-video-audio-track.html: Renamed from LayoutTests/media/media-video-audio-track.html.
* LayoutTests/platform/glib/TestExpectations:
* LayoutTests/platform/mac-wk1/TestExpectations:
* Source/WebCore/platform/graphics/avfoundation/objc/SourceBufferPrivateAVFObjC.h:
* Source/WebCore/platform/graphics/avfoundation/objc/SourceBufferPrivateAVFObjC.mm:
(WebCore::SourceBufferPrivateAVFObjC::precheckInitialisationSegment):
(WebCore::SourceBufferPrivateAVFObjC::didProvideMediaDataForTrackId):
(WebCore::SourceBufferPrivateAVFObjC::processFormatDescriptionForTrackId):
(WebCore::SourceBufferPrivateAVFObjC::didProvideContentKeyRequestInitializationDataForTrackID):
(WebCore::SourceBufferPrivateAVFObjC::appendInternal):
(WebCore::SourceBufferPrivateAVFObjC::destroyRenderers):
(WebCore::SourceBufferPrivateAVFObjC::clearTracks):
(WebCore::SourceBufferPrivateAVFObjC::trackDidChangeEnabled):
(WebCore::SourceBufferPrivateAVFObjC::setCDMInstance):
(WebCore::SourceBufferPrivateAVFObjC::flush):
(WebCore::SourceBufferPrivateAVFObjC::rendererWasAutomaticallyFlushed):
(WebCore::SourceBufferPrivateAVFObjC::trackIsBlocked const):
(WebCore::SourceBufferPrivateAVFObjC::canEnqueueSample):
(WebCore::SourceBufferPrivateAVFObjC::enqueueSample):
(WebCore::SourceBufferPrivateAVFObjC::isReadyForMoreSamples):
(WebCore::SourceBufferPrivateAVFObjC::didBecomeReadyForMoreSamples):
(WebCore::SourceBufferPrivateAVFObjC::notifyClientWhenReadyForMoreSamples):

Canonical link: https://commits.webkit.org/271655@main
